e2e/__tests__/nativeEsmWasm.test.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {resolve} from 'path';
+import {json as runJest} from '../runJest';
+
+const DIR = resolve(__dirname, '../native-esm-wasm');
+
+test('runs WASM test with native ESM', () => {
+  const {exitCode, json} = runJest(DIR, [], {
+    nodeOptions: '--experimental-vm-modules --no-warnings',
+  });
+
+  expect(exitCode).toBe(0);
+
+  expect(json.numTotalTests).toBe(4);
+  expect(json.numPassedTests).toBe(4);
+});
+
+test('runs WASM test with native ESM with --experimental-wasm-modules flag', () => {
+  const {exitCode, json} = runJest(DIR, [], {
+    nodeOptions:
+      '--experimental-vm-modules --experimental-wasm-modules --no-warnings',
```

Comment:
Interesting: CI has passed with and without `--experimental-wasm-modules`. I guess that in our case the usage of this flag is not necessary, because wasm imports are implemented via `new SyntheticModule(...)` (`--experimental-vm-modules`).

Let me try to get rid of the flag completely and merge `native-esm-wasm` with `native-esm` for simplicity.

Reply:
Done in https://github.com/facebook/jest/pull/13505/commits/dcff11e50e14934e548c2b08e8b45cb5e91968dd. Because `--experimental-wasm-modules` is not necessary, seems like we don’t even have to update the docs!

 Existing markdown still looks accurate: https://jestjs.io/docs/ecmascript-modules. It does not mention all supported cases, so not sure if wasm should be any special here.

Reply:
Awesome! It might make sense to mention that wasm doesn't need an _extra_ flag (at least at the moment - node might decide to unflag `vm` modules and at that point require it for wasm? unsure)

Reply:
Done in https://github.com/facebook/jest/pull/13505/commits/27d73b67eae02d27b94d9b6b62b8733ad9cf371e (happy to improve the docs further based on your suggestions).